### PR TITLE
[FSDP()][12/N] Easy cleanup

### PR DESCRIPTION
**Pull request by awgu**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #88123 [FSDP] Rename `unflat_param_name` -> `fqn` for consistency
* #88122 [FSDP] Simplify `_get_buffer_names()`
* #88121 [FSDP] Remove unneeded `torch.no_grad()` context when offloading to CPU
* #88120 [FSDP][Docs] Add note mentioning rate limiter for backward prefetch
* #88040 [FSDP()][27/N] Add forward hook registration
* #87941 [FSDP()][26/N] Move `_lazy_init()` into `_fsdp_root_pre_forward()`
* #87940 [FSDP()][25/N] Add `_post_forward_reshard()`
* #87939 [FSDP()][24/N] Refactor `_lazy_init()`
* #87938 [FSDP()][23/N] Refactor handle attr initialization
* #87937 [FSDP] Simplify `_reset_lazy_init()`
* #87936 [FSDP()][22/N] Refactor `_cast_buffers()` in `_lazy_init()`
* #87935 [FSDP()][21/N] Refactor `_buffer_name_to_orig_dtype` computation
* #87934 [FSDP] Rename `dtype` to `buffer_name_to_dtype`
* #87933 [FSDP] Remove `device` arg from `_cast_buffers()`
* #87932 [FSDP()][20/N][Easy] Move functions in file
* #87931 [FSDP()][18/N] Refactor `pre_forward_unshard()`
* #87930 [FSDP()][17/N] Refactor `_fsdp_root_pre_forward()`
* #87929 [FSDP()][16/N] Refactor post-forward/pre-backward
* #87928 [FSDP()][15/N] Refactor `_init_streams()`
* #87927 [FSDP()][14/N] Refactor pre-forward/post-backward
* #87926 [FSDP()][13/N] Refactor unshard/reshard/grads
* **#87925 [FSDP()][12/N] Easy cleanup**
* #87924 [FSDP()][10/N][11/N] Introduce composable (ctor only)
* #87923 [FSDP()][9/N] Refactor ctor (continued)

### CODE REVIEW

File: torch/distributed/fsdp/fully_sharded_data_parallel.py [hunk 1]
```diff
@@ -532,11 +525,6 @@ def __init__(
         )
         self.register_load_state_dict_post_hook(_post_load_state_dict_hook)
 
-    def _register_param_handle(self, handle: FlatParamHandle) -> None:
```
Comment (awgu) [inline torch/distributed/fsdp/fully_sharded_data_parallel.py:535]: Unused.

File: torch/distributed/fsdp/_init_utils.py [hunk 2]
```diff
@@ -124,6 +124,11 @@ def _init_core_state(
     backward_prefetch_limit: int,
     forward_prefetch_limit: int,
 ) -> _State:
+    # We clamp the strategy to `NO_SHARD` for world size of 1 since they are
```
Comment (awgu) [inline torch/distributed/fsdp/_init_utils.py:127]: Simple move.

File: torch/distributed/fsdp/fully_sharded_data_parallel.py [hunk 3]
```diff
@@ -516,9 +511,7 @@ def __init__(
             _check_orig_params_flattened(self, self._ignored_params)
             self._register_flat_param()
 
-        # TODO (revisit): I explicitly delete this because we do want to keep
-        # references to these from FSDP. I only added it to the state for
-        # convenience in this refactoring.
+        # Delete to avoid keeping references after the constructor
```
Comment (awgu) [inline torch/distributed/fsdp/fully_sharded_data_parallel.py:514]: We can revisit if we want to get around this `delattr` as a follow-up.

Comment (mrshenli) [reply torch/distributed/fsdp/fully_sharded_data_parallel.py:514]: This makes sense to me

**Review by mrshenli — APPROVED** [review]

**Pull request closed without merging**